coldfront/core/project/views_/removal_views.py

```diff
@@ -443,87 +438,40 @@ def form_valid(self, form):
+                    ProjectUserRemovalRequestStatusChoice.objects.get(
+                        name=status)
+                request_obj.save()
+                if status == 'Complete':
+                    request_obj.completion_time = utc_now_offset_aware()
+                    request_obj.save()
+                    # Run the runner as the last step of the transaction, to
+                    # avoid writing to the log and sending emails for failed
+                    # transactions.
+                    runner = ProjectRemovalRequestProcessingRunner(request_obj)
+                    runner.run()
+        except Exception as e:
+            message = f'Rolling back failed transaction. Details:\n{e}'
+            logger.exception(message)
+            message = (
+                'Unexpected failure. Please try again, or contact an '
+                'administrator if the problem persists.')
+            messages.error(self.request, message)
+        else:
+            message = (
+                f'Project removal request initiated by '
+                f'{request_obj.requester.username} for User '
+                f'{self.user_obj.username} under Project '
+                f'{request_obj.project_user.project.name} is complete.')
+            messages.success(self.request, message)
 
-        if EMAIL_ENABLED and status == 'Complete':
-            pi_condition = Q(
-                role__name='Principal Investigator', status__name='Active',
-                enable_notifications=True)
-            manager_condition = Q(role__name='Manager', status__name='Active')
-            manager_pi_queryset = project_obj.projectuser_set.filter(
-                pi_condition | manager_condition)
-
-            for proj_user in list(chain(manager_pi_queryset, [removed_user])):
-                curr_user = proj_user.user
-                template_context = {
-                    'user_first_name': curr_user.first_name,
-                    'user_last_name': curr_user.last_name,
-                    'removed_user_first_name': removed_user.user.first_name,
-                    'removed_user_last_name': removed_user.user.last_name,
-                    'requester_first_name': requester.first_name,
-                    'requester_last_name': requester.last_name,
-                    'project_name': project_obj.name,
-                    'signature': EMAIL_SIGNATURE,
-                    'support_email': SUPPORT_EMAIL,
-                }
-
-                send_email_template(
-                    'Project Removal Request Completed',
-                    'email/project_removal/project_removal_complete.txt',
-                    template_context,
-                    EMAIL_SENDER,
-                    [curr_user.email]
-                )
+            if isinstance(runner, ProjectRemovalRequestProcessingRunner):
+                for message in runner.get_warning_messages():
+                    messages.warning(self.request, message)
```

Comment:
Do we want to display messages about database objects to the user? Although I guess the only people that would see them are admins.

Reply:
I skipped those because they're lower-level details that admins shouldn't be concerned with. (They get logged.)